app/src/main/java/com/sdp/movemeet/Backend/FirebaseInteraction.java

```diff
@@ -29,14 +30,32 @@
 import java.util.HashMap;
 import java.util.Map;
 
+/**
+ * This class allows various types of interaction with Firestore services (Firestore, Storage and
+ * Authentication) to both save and fetch information about the user. It also allows to store and
+ * retrieve images (user profile picture and activity header picture) by interacting with Firebase
+ * Storage and avoid code duplicates through diverse Activity.
+ */
 public class FirebaseInteraction {
```

Comment:
This whole class should also be rewritten to use the new generic interface; I'll take care of that

Reply:
I can help ! Mainly with the methods for saving and loading images, as they will be correlated to the cache

Reply:
Yes, as said by Dorian, this class should be entirely removed in a near future!